pkg/resources/rbac.authorization.k8s.io/v1/rolebinding/validator.go

```diff
@@ -0,0 +1,122 @@
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/trace"
+)
+
+const (
+	grbOwnerLabel = "authz.management.cattle.io/grb-owner"
+)
+
+// Validator implements admission.ValidatingAdmissionHandler.
+type Validator struct {
+	admitter admitter
+}
+
+// NewValidator returns a new validator for rolebindings.
+func NewValidator() *Validator {
+	return &Validator{
+		admitter: admitter{},
+	}
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group:    "rbac.authorization.k8s.io",
+		Version:  "v1",
+		Resource: "rolebindings",
+	}
+}
+
+// Operations returns list of operations handled by this validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{
+		// admissionv1.Create, TODO Not sure if needed
+		admissionregistrationv1.Update,
+	}
+}
+
+// ValidatingWebhook returns the ValidatingWebhook used for this CRD.
+func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) []admissionregistrationv1.ValidatingWebhook {
+
+	webhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.NamespacedScope, v.Operations())
+
+	webhook.ObjectSelector = &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      grbOwnerLabel,
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	}
+
+	return []admissionregistrationv1.ValidatingWebhook{*webhook}
+}
+
+// Admitters returns the admitter objects used to validate roles.
+func (v *Validator) Admitters() []admission.Admitter {
+	return []admission.Admitter{&v.admitter}
+}
+
+type admitter struct{}
+
+// Admit is the entrypoint for the validator. Admit will return an error if it's unable to process the request.
+// If this function is called without NewValidator(..) calls will panic.
+func (a *admitter) Admit(request *admission.Request) (*admissionv1.AdmissionResponse, error) {
+	listTrace := trace.New("rolebindingValidator Admit", trace.Field{Key: "user", Value: request.UserInfo.Username})
+	defer listTrace.LogIfLong(admission.SlowTraceDuration)
+
+	oldRoleBinding, newRoleBinding, err := objectsv1.RoleBindingOldAndNewFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	fldPath := field.NewPath("rolebinding")
+
+	fieldErr := validateGRBOwnerLabel(oldRoleBinding, newRoleBinding, fldPath)
+	if fieldErr != nil {
+		return admission.ResponseBadRequest(fieldErr.Error()), nil
+	}
+
+	return admission.ResponseAllowed(), nil
+}
+
+// validateGRBOwnerLabel checks that a user isn't removing or modifying the grbOwnerLabel.
+// Users can add the label if it didn't already exist.
+func validateGRBOwnerLabel(old, new *v1.RoleBinding, fldPath *field.Path) *field.Error {
```

Comment:
This logic is very similar to the logic for the grOwnerLabel (and will probably be something that we use in the future). You might want to consider moving it to `pkg/resources/common`